Sources/TestHelpers/URLRequestSnapshot.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  SnapshotStrategy.swift
+//  Supabase
+//
+//  Created by Guilherme Souza on 22/01/25.
+//
+
+@preconcurrency import InlineSnapshotTesting
+
+#if !os(WASI)
+  import Foundation
+
+  #if canImport(FoundationNetworking)
+    import FoundationNetworking
+  #endif
+
+  extension Snapshotting where Value == URLRequest, Format == String {
+    /// A snapshot strategy for comparing requests based on a cURL representation.
+    ///
+    // ``` swift
+    // assertSnapshot(of: request, as: .curl)
+    // ```
+    //
+    // Records:
+    //
+    // ```
+    // curl \
+    //   --request POST \
+    //   --header "Accept: text/html" \
+    //   --data 'pricing[billing]=monthly&pricing[lane]=individual' \
+    //   "https://www.pointfree.co/subscribe"
+    // ```
+    package static let _curl = SimplySnapshotting.lines.pullback { (request: URLRequest) in
```

Comment:
As discussed in https://github.com/pointfreeco/swift-snapshot-testing/discussions/945, I had to write our own snapshot strategy that uses both `httpBody` and `httpBodyStream`, as this won't be merged upstream.